checker.go

```diff
@@ -341,18 +344,19 @@ func (n numArgs) NumArgs() int {
 func match(got string, pattern interface{}, msg string) error {
 	regex, ok := pattern.(string)
 	if !ok {
-		return BadCheckf("the regular expression pattern must be a string, got %T instead", pattern)
+		return testerror.BadCheckf(
+			"the regular expression pattern must be a string, got %T instead", pattern)
 	}
 	matches, err := regexp.MatchString("^("+regex+")$", got)
 	if err != nil {
-		return BadCheckf("cannot compile regular expression %q: %s\n", regex, err)
+		return testerror.BadCheckf("cannot compile regular expression %q: %s\n", regex, err)
```

Comment:
I don't think you want newlines in your error messages.

Reply:
Why not?